src/actions/dataset.js

```diff
@@ -44,6 +47,32 @@ export const getDatasets = () => (dispatch) => {
   }
 };
 
+export const addDataset = ({
+  fileCustomName,
+  fileLocation,
+  fileDescription,
+}) => (dispatch) => {
+  try {
```

Comment:
You might want to add flags here to trigger the spinner to wait for the file to correctly load.